tests/integration/memory_resource_traits_tests.cpp

```diff
@@ -7,129 +7,96 @@
-
-  auto alloc_one = rm.getAllocator("HOST");
-  auto alloc_two = rm.makeAllocator<umpire::strategy::SizeLimiter>(
-      "HOST_Limiter", alloc_one, 1024);
-
-  double* data =
-      static_cast<double*>(alloc_one.allocate(1024 * sizeof(double)));
-
-  ASSERT_EQ(MemoryResourceTraits::resource_type::host,
-            alloc_two.getAllocationStrategy()->getTraits().resource);
-
-  ASSERT_THROW(alloc_two.deallocate(data), umpire::util::Exception);
-
-  ASSERT_NO_THROW(alloc_one.deallocate(data));
+  if(resource == "HOST")
+    return myResource::host;
+  else if(resource == "DEVICE")
+    return myResource::device;
+  else if(resource == "DEVICE_CONST")
+    return myResource::device_const;
+  else if(resource == "UM")
+    return myResource::um;
+  else if(resource == "PINNED")
+    return myResource::pinned;
+  else if(resource == "FILE")
+    return myResource::file;
+  else //unknown
+    return myResource::unknown;
 }
 
-#if defined(UMPIRE_ENABLE_FILE_RESOURCE)
-TEST(MemoryResourceTraitsTest, FILE_Resource)
+TEST_P(MemoryResourceTraitsTest, ResourceTraitTest)
 {
-  auto& rm = umpire::ResourceManager::getInstance();
-
-  auto alloc_one = rm.getAllocator("FILE");
-  auto alloc_two = rm.makeAllocator<umpire::strategy::SizeLimiter>(
-      "FILE_Limiter", alloc_one, 1024);
-
   double* data =
-      static_cast<double*>(alloc_one.allocate(1024 * sizeof(double)));
-
-  ASSERT_EQ(MemoryResourceTraits::resource_type::file,
-            alloc_two.getAllocationStrategy()->getTraits().resource);
-
-  ASSERT_THROW(alloc_two.deallocate(data), umpire::util::Exception);
+      static_cast<double*>(m_allocator->allocate(1024 * sizeof(double)));
 
-  ASSERT_NO_THROW(alloc_one.deallocate(data));
-}
-#endif
-#if defined(UMPIRE_ENABLE_DEVICE)
-TEST(MemoryResourceTraitsTest, DEVICE_Resource)
-{
-  auto& rm = umpire::ResourceManager::getInstance();
-
-  auto alloc_one = rm.getAllocator("DEVICE");
-  auto alloc_two = rm.makeAllocator<umpire::strategy::SizeLimiter>(
-      "DEVICE_Limiter", alloc_one, 1024);
+  myResource resource = getResourceType(m_resource);
 
-  double* data =
-      static_cast<double*>(alloc_one.allocate(1024 * sizeof(double)));
+  ASSERT_EQ(resource, m_allocator_pool->getAllocationStrategy()->getTraits().resource);
 
-  ASSERT_EQ(MemoryResourceTraits::resource_type::device,
-            alloc_two.getAllocationStrategy()->getTraits().resource);
+  ASSERT_EQ(m_allocator->getName(), m_resource);
 
-  ASSERT_THROW(alloc_two.deallocate(data), umpire::util::Exception);
+  ASSERT_THROW(m_allocator_pool->deallocate(data), umpire::util::Exception);
 
-  ASSERT_NO_THROW(alloc_one.deallocate(data));
+  ASSERT_NO_THROW(m_allocator->deallocate(data));
 }
+
+//returns a vector of strings with the names of the
+//memory resources currently available.
+std::vector<std::string> memory_resource_strings()
+{
+  std::vector<std::string> resources;
+  resources.push_back("HOST");
+#if defined(UMPIRE_ENABLE_DEVICE)
+  resources.push_back("DEVICE");
```

Comment:
I think you want to add a loop over number devices here so we can test all of the available devices.

Reply:
It will give an error if I do that.. I forget what exactly it is (I can reproduce it and remind myself), but I thought it had something to do with peer devices

Reply:
That was in your accessibility branch iirc.